### ISSUE TITLE
GUI scripts should use pythonw not python on windows

### ISSUE THREAD

**Issue by mwhapples** [pdm-project/pdm#1804]
- [x] I have searched the issue tracker and believe that this is not a duplicate.

**Make sure you run commands with `-v` flag before pasting the output.**

## Steps to reproduce

<!--Describe the minimized example of how to reproduce the bug-->
Following steps to be carried out on windows:
* Using the attached example project.
* Run pdm install to get it in the venv. 
* In Windows explorer navigate to the venv location, go to scripts and run the GUI scripts wrapper .exe
## Actual behavior

<!--A clear and concise description the result of the above steps-->

As well as any GUI windows appearing there will be a console window launched. The python executable used is python.exe as shown in the label in the GUI.
## Expected behavior

<!--A clear and concise description of what you expected to happen.-->
Only the scripts GUI windows should appear. The python executable should be pythonw.exe as shown in the GUI label.
## Environment Information

```bash
# Paste the output of `pdm info && pdm info --env` below:
PDM version:                                                                                                                         
  2.5.0b0                                                                                                                            
Python Interpreter:                                                                                                                  
  C:\Users\Michael\gui-example\.venv\Scripts\python.exe (3.11)                                                                       
Project Root:                                                                                                                        
  C:/Users/Michael/gui-example                                                                                                       
Local Packages:                                                                                                                      
                                                                                                                                     
{                                                                                                                                    
  "implementation_name": "cpython",                                                                                                  
  "implementation_version": "3.11.2",                                                                                                
  "os_name": "nt",                                                                                                                   
  "platform_machine": "AMD64",                                                                                                       
  "platform_release": "10",                                                                                                          
  "platform_system": "Windows",                                                                                                      
  "platform_version": "10.0.22621",                                                                                                  
  "python_full_version": "3.11.2",                                                                                                   
  "platform_python_implementation": "CPython",                                                                                       
  "python_version": "3.11",                                                                                                          
  "sys_platform": "win32"                                                                                                            
}                                                                                                                                    
 
```
[gui-example.zip](https://github.com/pdm-project/pdm/files/11132415/gui-example.zip)

**Comment by frostming** [comment]
Can you help to confirm that `pythonw.exe` always co-exists with `python.exe`, even in a virtualenv created by both `virtualenv` and `venv`?

**Comment by mwhapples** [comment]
Looking at pep405 on virtual environments it says nothing about pythonw.exe whether this is an oversight or intentional I do not know.
In my experience with venv it seems like if pythonw is in the python installation then it is included. I have tried CPython from python.org, pypy and conda distributions of python on Windows. One case I have encountered which seems to differ from this is with the python included in msys2/MingW, here python does not include pythonw.exe in the base install and so not in the virtual environment it creates, but I think this is a special case as this does not report sys.platform as win32 but rather cygwin.
Also the python docs on using python on Windows does mention the pythonw.exe executable https://docs.python.org/3/using/windows.html
As I don't use old python versions which do not include the venv module, I have no experience with virtualenv, I would need to check that one.
In the entrypoints specification of the python packaging documentation https://packaging.python.org/en/latest/specifications/entry-points/ it does mention how GUI scripts should be wrapped in an GUI executable.
"The difference between console_scripts and gui_scripts only affects Windows systems. console_scripts are wrapped in a console executable, so they are attached to a console and can use sys.stdin, sys.stdout and sys.stderr for input and output. gui_scripts are wrapped in a GUI executable, so they can be started without a console, but cannot use standard streams unless application code redirects them. Other platforms do not have the same distinction."
